addons/mass_mailing/models/res_company.py

```diff
@@ -7,11 +7,13 @@
 class ResCompany(models.Model):
     _inherit = "res.company"
 
-    def _get_social_media_links(self):
-        self.ensure_one()
-        return {
-            'social_facebook': self.social_facebook,
-            'social_linkedin': self.social_linkedin,
-            'social_twitter': self.social_twitter,
-            'social_instagram': self.social_instagram
-        }
+    def try_set_social_media_links(self, values):
+        fields = ['social_facebook', 'social_youtube', 'social_linkedin', 'social_linkedin', 'social_instagram']
+
+        link_to_save = {}
+        for field in fields:
+            if (values.get(field)):
+                link_to_save[field] = values.get(field)
+
+        if link_to_save:
+            self.env.company.write(link_to_save)
```

Comment:
This entire thing seems completely unnecessary and to have no added value, why can't the client just call `write()` directly?

Also why does it not allow *unsetting* a link? What if you close your youtube page or get banned from facebook?